src/solvers/td3.ml

```diff
@@ -627,22 +629,40 @@ module WP =
             )
         in
 
+        let force_reanalyze = StringSet.of_list @@ GobConfig.get_string_list "incremental.force-reanalyze.funs" in
+        let reluctant = GobConfig.get_bool "incremental.reluctant.on" in
+        let reanalyze_entry f =
+          (* destabilize the entry points of a changed function when reluctant is off,
+             or the function is to be force-reanalyzed  *)
+          (not reluctant) || StringSet.mem f.svar.vname force_reanalyze
+        in
+
         let obsolete_ret = HM.create 103 in
         let obsolete_entry = HM.create 103 in
         let obsolete_prim = HM.create 103 in
+
+        (* When reluctant is on:
+           Only add function entry nodes to obsolete_entry if they are in force-reanalyze *)
         List.iter (fun f ->
-            mark_node obsolete_entry f (FunctionEntry f);
-            mark_node obsolete_ret f (Function f);
+            if reanalyze_entry f then
+              (* collect function entry for eager destabilization *)
+              mark_node obsolete_entry f (FunctionEntry f)
+            else
+              (* collect function return for reluctant analysis *)
+              mark_node obsolete_ret f (Function f)
           ) changed_funs;
-        List.iter (fun (f, pn, _) ->
-            List.iter (fun n ->
-                mark_node obsolete_prim f n
-              ) pn;
-            mark_node obsolete_ret f (Function f);
-          ) part_changed_funs;
+        (* Unknowns from partially changed functions need only to be collected for eager destabilization when reluctant is off *)
```

Comment:
But also when the partially changed function is force reanalyzed, no?

Reply:
Force-reanalyzed functions are not (and should not) be treated as partially changed: 
https://github.com/goblint/analyzer/blob/304cfd62b978352c62fbc74dabc975971ca696fb/src/incremental/compareCIL.ml#L37-L40

Reply:
Right. I guess what makes it confusing is how `incremental.force-reanalyze.funs` is handled in two totally different places now. The suggestion from https://github.com/goblint/analyzer/pull/600#discussion_r809787070 might help.

Reply:
Now the list of `varinfo`s of functions in force-reanalyze is also collected in a `VarinfoSet` in `CompareCIL`, in the `change_info`. So the solver does not have to query that option here again.
These force-reanalyze functions are also still collected in the `changed` field of the `change_info`, because otherwise one would have to treat them separately everywhere where the `changed` set is used (e.g. in `UpdateCIL` or the solver).